src/main/java/org/truffleruby/core/rope/TruffleRopesNodes.java

```diff
@@ -135,7 +135,9 @@ private static String getStructure(ConcatRope rope) {
         }
 
         private static String getStructure(SubstringRope rope) {
-            return getStructure(rope.getChild()) + "[" + rope.getByteOffset() + ", " + rope.characterLength() + "]";
+            final Rope child = rope.getChild();
+            final int characterOffset = RopeOperations.strLength(child.getEncoding(), child.getBytes(), 0, rope.getByteOffset());
```

Comment:
Is there a reason to not use a `StringByteCharacterIndexNode` child here? I don't think we want to be spreading this logic about.